Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift

```diff
@@ -118,28 +117,32 @@ private final class HTTPClientUpgradeHandler<Upgrader>: ChannelInboundHandler wh
             return
         }
 
-        // We're trying to remove ourselves from the pipeline but not upgrading, so just pass this on.
-        if seenFirstResponse {
-            ctx.fireChannelRead(data)
-            return
-        }
-
         let responsePart = unwrapInboundIn(data)
-        seenFirstResponse = true
 
         // We should only ever see a request header: by the time the body comes in we should
         // be out of the pipeline. Anything else is an error.
-        guard case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) else {
+        if case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) {
+            self.res = res
+        } else if case .end(_) = responsePart, let res = self.res {
+            upgrading = true
+            
+            removeExtraHandlers(ctx: ctx).then {
+                return self.upgrader.upgrade(ctx: ctx, upgradeResponse: res)
+                }.then { handler in
+                    let p: EventLoopPromise<Bool> = ctx.eventLoop.newPromise()
+                    let resultFuture = p.futureResult.map { _ -> Upgrader.UpgradeResult in
```

Comment:
are we OK ignoring the `Bool` value here?

Reply:
Yeah, the bool will only ever be `true`, as you can see in https://github.com/apple/swift-nio/blob/nio-1.12/Sources/NIO/ChannelPipeline.swift#L507

Reply:
To be sure, I asked the NIO guys: https://discordapp.com/channels/431917998102675485/448584561845338139/534780196516134912

Reply:
PR up on the NIO side. Will be shipped in NIO 2 :)